lib/open_api_spex/plug/cast_and_validate.ex

```diff
@@ -95,28 +92,24 @@ defmodule OpenApiSpex.Plug.CastAndValidate do
           private: %{
             phoenix_controller: controller,
             phoenix_action: action,
-            open_api_spex: private_data
+            open_api_spex: _
           }
         },
         opts
       ) do
+    {_spec, operation_lookup} = OpenApiSpex.Plug.Cache.spec_and_operation_lookup(conn)
+
     operation =
-      case private_data.operation_lookup[{controller, action}] do
+      case operation_lookup[{controller, action}] do
         nil ->
-          operationId = controller.open_api_operation(action).operationId
-          operation = private_data.operation_lookup[operationId]
-
-          operation_lookup =
-            private_data.operation_lookup
-            |> Map.put({controller, action}, operation)
+          operation_id = controller.open_api_operation(action).operationId
 
-          OpenApiSpex.Plug.Cache.adapter().put(
-            private_data.spec_module,
-            {private_data.spec, operation_lookup}
+          OpenApiSpex.Plug.Cache.get_and_cache_controller_action(
```

Comment:
Why is it necessary to also cache the operation under `{controller, action}`? It's already cached under `operation_id`, which is one-to-one with `{controller, action}`. I tried removing the `{controller, action}` caching, and all the tests passed, and all the tests also passed in a codebase that uses open_api_spex.

Reply:
Caching by `{controller, action}` prevents calling into the controllers `open_api_operation/1` function on every request, just to get the operationId:

```elixir
operation_id = controller.open_api_operation(action).operationId
```
When using `@doc` based specs, that call includes rebuilding the `%Operation{}` from the doc metadata.
I guess it just _feels_ wasteful to rebuild all that structure, only to throw most of it away :)

Reply:
Ok, I see that now.